Eastern-Research-Group/mywaterway
# Feature/461 update dependencies

## Related Issues:
* [HMW-461](https://jira.epa.gov/browse/HMW-461)

## Main Changes:
* Updated 3rd party dependencies.
  * Skipping React 18 again. Also skipped `react-sticky-box` since it requires React 18.
* Fixed an issue where zoom changes would hide layers added via the Add/Save Data Widget.

## Steps To Test:
1. Run `npm ci --legacy-peer-deps` at the app and app/client level
2. Run `npm ci` at the app/server levels
3. Add a few layers in the add data widget
4. Click the layers button in the Add/Save Data Widget
5. Click the zoom button next to some of the layers
6. Verify the map zooms to the layer and that the layer is visible in the layer list
  * Note: Some layers may not be visible on the map after clicking zoom, because they zoom out to the entire globe which is outside of the layer's visible scale range.
7. Test as many other features as possible
    * Test the highcharts downloads. It looks like the latest version of highcharts fixed the download issues we were seeing before, where some download formats didn't work.
    * Verify that refreshing a page zooms to where it should on the map. I noticed an issue after the update where if I refreshed the community page it would zoom to the entire US, instead of the HUC boundaries. Similar issues occured on the monitoring report, plan summary, tribal, and waterbody report pages. What I did was add some code to verify the mapView is ready to go using `mapView.when`.
